test/NSHM_NZ/inversion/BuildCrustalFaultSubsectionsTest.java

```diff
@@ -0,0 +1,136 @@
+ *          
+ * @author Thingbaijam        
+ * 
+ * Note to myself:
+ * 
+ * getSubSectionsList() is declared (overloaded) in the interface FaultSection and implemented in class FaultSectionPrefData.
+ * Class DeformationModelFetcher also has a method of same name, which ultimately can be connected to the implementation in FaultSectionPrefData.
+ * 
+ *    
+*/
+
+
+public class BuildCrustalFaultSubsectionsTest {
+
+	private static final double grid_disc = 5d;
+
+	
+	static List<FaultSection> fsd;
+	
+	@BeforeClass
+	public static void setUpBeforeClass() throws Exception {
+		// this is the input fault section data file
+		// Still use alderman_sections.xml
+		
+		File fsdFile = new File("./test/NSHM_NZ/inversion/fixtures/alderman_sections.xml");
+		
+		
+		// load in the fault section data ("parent sections")
+		// Note2self: Is this natural way to load the fault sections (or fault models). See class ClusterRuptureBuilder to understand better
+		// Likewise, hard-codes in emums FaultModels, DeformationModels, of course - these are in package UCERF3. 
+		// perhaps, a work-sheet could be use to describe/read in the models  
+		fsd = FaultModels.loadStoredFaultSections(fsdFile);
+	}
+	
+    @Test 
+	public void testSubsection() throws IOException {
+		
+		System.out.println(fsd.size() + " Crustal Fault Sections");
+		// this list will store our subsections
+		List<FaultSection> subSections = Lists.newArrayList();
+		
+		// sub section length factor (in terms of DDW)
+		double SubSectionLengthFactor = 0.5;
+
+		//  Build and inspect the subsections 
+		int sectIndex = 0;
+		
+		for (FaultSection pSect : this.fsd) {
+			
+			double ddw = pSect.getOrigDownDipWidth();
+			
+			double maxSectLength = ddw*SubSectionLengthFactor; // why is this the maximum?
+			
+			// the "2" here sets a minimum number of sub sections
+			List<? extends FaultSection> newSubSects = pSect.getSubSectionsList(maxSectLength, sectIndex, 1);
+			
+			// Note2myself: fundamentally, this computation is done by getEqualLengthSubsectionTraces() defined in class FaultUtils  
+			System.out.println("Fault Section = " + pSect.getName() + " ------------------");
+			System.out.println("0.5 DDW = "+ 0.5*ddw);
+			
+			
+			// Two ways by which we can retrieve the length of a subsection, which provide approximately equal lengths. 
+			// I assume that a Fault Trace is a straight line. 
+			// double lengthOfSubSection =  pSect.getFaultTrace().getTraceLength()/newSubSects.size();
+			// System.out.println("Length of subsection by division of the fault trace = " 
+			//			+ lengthOfSubSection);
+			
+			double lengthOfSubSection = newSubSects.get(1).getFaultTrace().getTraceLength();
+			
+			System.out.println("Length of subsection from the subsection = " + lengthOfSubSection);
+			
+			assertEquals(Boolean.TRUE,lengthOfSubSection>=0.5*ddw);
+			// if it fails, length of at least two subsections may be smaller than DDW   
```

Comment:
Does this test fail on some test data?? If the point is to show that it does in some cases, then best create a test to explain that particular case.

Reply:
It fails all the time. The reason is length of subsection is never equal to (but always smaller) than half of the seismogenic width. However,  documentation (UCERF3) report says "seismogenic depth" - perhaps, a small change using depth (lower - upper)  instead of DDW could be useful.